### DISCUSSION TITLE
Tuya based IR bridge LED

### DISCUSSION THREAD

**Discussion by chrwei** [arendst/Tasmota#14574] [Q&A] [answered]
I have tasmota_ir working, but I want to turn that little blue LED off.  The stock firmware had it off, so I know it's technically possible.

I'd really like if it blinked when an IR command is sent or received, but I'd settle for "off when everything is OK" and blinking constand when there's wifi or mqtt issues.  

Can this be done with a config option, or even a console or mqtt command?

**Comment by sfromis** [comment, marked as answer]
Depends on which device it is, which template it is, which gpios are configured how.

**Comment by chrwei** [comment]
Tuya device.  https://tasmota.github.io/docs/devices/YTF-IR-Bridge/

GPIO4 is what I keep seeing but i don't have a way to confirm that.  I've tried editing the template to change GPIO4 from LED_i User and PWM but nothing seems to make a difference

**Reply by barbudor** [reply]
Where is the led ?
Can you trace the PCB from the ESP module to the led ?
If the reference model had a blue led, it would probably have been listed in the template

Or did you replaced an incompatible module with en ESP-12 ?
AI Thinker's ESP12S has a blue led on GPIO2
![image](https://user-images.githubusercontent.com/2996042/150697146-93956703-7403-43a6-b303-de2b301e3b31.png)

**Reply by chrwei** [reply]
I opened it again, it's a TYWE3S, Tuya's ESP8266EX module.  based on pinouts I found online, I get continuity from GPIO4 to the test point near the LED, and about 190K ohm to the LED, which seems about right.  

What I can't seem to make work is setting GPIO4 LOW to turn it off

**Reply by sfromis** [reply]
According to the link you posted, the blue led is indeed gpio4. If you configure it as a relay, you can check that by toggling it on and off. If relevant, you can use the same method to hunt for it on other pins.

**Reply by barbudor** [reply]
190K is to much. 190 ohms would be logical

Setting it at LedLink (or LedLink_i) should provide you what you are looking for (regarding Wifi and MQTT, not blinking when IR is set or received)

If you suceed in controlling it, you could make it blink by setting it as Relay or Relay_I, use Pulsetime to set a short pulse and use rules to trigger the pulse when receiving an IRsend or when sending an IRReceive

**Reply by chrwei** [reply]
I change the template to Relay, LED is still on.  I issue Power on the console and it just says error.

Everything I can find on how to toggle outputs just isn't.  Is this stripped from the IR firmware or what am I missing?

**Reply by sfromis** [reply]
If the Power command says error, you have not configured a relay, details depend on you you did the device configuration. You can check the outcome of the config using the command `gpio all`, which will tell you the config status of all gpio pins.

**Reply by chrwei** [reply]
ah ha!  that at least helped me track down the issue.

I had assumed that editing the template edits the active template, it does not.  the active was the defail IR Bridge (62), so my edits did not apply.

So, at least now I can control it manually.  it still be nice if I just leave as LED and turn it off at will so the wifi and mqtt issues can still make it blink

**Reply by sfromis** [reply]
If you want to play with it.... https://tasmota.github.io/docs/Lights/#status-leds

**Reply by chrwei** [reply]
odd.  setting it back to LED it doesn't come back on.  I think I'll leave it and see what happens
